llvm/lib/Support/Windows/Path.inc

```diff
@@ -154,7 +154,10 @@ std::string getMainExecutable(const char *argv0, void *MainExecAddr) {
     return "";
 
   llvm::sys::path::make_preferred(PathNameUTF8);
-  return std::string(PathNameUTF8.data());
+
+  SmallString<256> RealPath;
+  sys::fs::real_path(PathNameUTF8, RealPath);
+  return (std::string)RealPath;
```

Comment:
Please use `.str()` rather than a C-style cast.

Reply:
> Please use `.str()` rather than a C-style cast.

I used `(std::string)` because `SmallString` has a `explicit operator std::string() const`.
You meant I should use `RealPath.str().str()`? (first `.str()` gives a `StringRef`, the second a `std::string`).

Reply:
In this case `std::string(RealPath)` should look better.